src/back-end/lib/permissions.ts

```diff
@@ -740,6 +741,50 @@ export async function editTWUOpportunity(
   );
 }
 
+/**
+ * Admins and proposal authors can edit Proposals, so long as they are in a
+ * certain state
+ *
+ * @param connection
+ * @param session
+ * @param proposalId
+ * @param opportunity
+ */
+export async function editTWUProposal(
+  connection: Connection,
+  session: Session,
+  proposalId: string,
+  opportunity: TWUOpportunity
+): Promise<boolean> {
+  return (
+    isAdmin(session) ||
+    (session &&
+      (await isTWUProposalAuthor(connection, session.user, proposalId)) &&
+      (await hasAcceptedPreviousTerms(connection, session))) ||
+    (session &&
+      (await isTWUOpportunityAuthor(
+        connection,
+        session.user,
+        opportunity.id
+      )) &&
+      doesTWUOpportunityStatusAllowGovToViewProposals(opportunity.status)) ||
+    false
+  );
+}
+
+export async function submitTWUProposal(
+  connection: Connection,
+  session: Session,
+  proposalId: string
+): Promise<boolean> {
+  return (
+    (session &&
+      (await isTWUProposalAuthor(connection, session.user, proposalId)) &&
+      (await hasAcceptedCurrentTerms(connection, session))) ||
+    false
```

Comment:
Do an organization check for correct TWU qualification.

Reply:
leaving for Ian in another PR